keyboards/keebit/coral/coral.c

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+// ===== LED/RGB INDICATOR =====
+enum RGB_LAYERS {
+    RGBL_CAPSL = 0,
+    RGBL_CAPSW = 1,
+};
+
+const rgblight_segment_t PROGMEM capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, RGBLED_NUM, HSV_BLUE}
+);
+
+const rgblight_segment_t PROGMEM capsword_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, RGBLED_NUM, HSV_WHITE}
+);
+
+// Now define the array of layers. Later layers take precedence
+const rgblight_segment_t* const PROGMEM rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+   [RGBL_CAPSL] =  capslock_layer,
+   [RGBL_CAPSW] =  capsword_layer
+);
+
+bool led_update_user(led_t led_state) {
+    rgblight_set_layer_state(RGBL_CAPSL, led_state.caps_lock);
+    return true;
+}
+
+void caps_word_set_user(bool active) {
+    rgblight_set_layer_state(RGBL_CAPSW, active);
+}
+
+void keyboard_post_init_user(void) {
+    // Enable the LED layers
+    rgblight_layers = rgb_layers;
+}
```

Comment:
KB level functions must be used at the kb level, not the user level functions.

Unfortunately, there is no user level version for caps word, so setting it as weak so users can override it is a decent workaround. 

```suggestion
bool led_update_kb(led_t led_state) {
    if (!led_update_user(led_state)) { 
        return false; 
    }
    rgblight_set_layer_state(RGBL_CAPSL, led_state.caps_lock);
    return true;
}

__attribute__((user)) void caps_word_set_user(bool active) {
    rgblight_set_layer_state(RGBL_CAPSW, active);
}

void keyboard_post_init_kb(void) {
    // Enable the LED layers
    rgblight_layers = rgb_layers;
    keyboard_post_init_user();
}
```

Reply:
I was about to change this as previously I wasn't aware of the `*_kb` functions and have no idea about how to set kb level caps word 😅